### PR TITLE
VOLDEV-165

### PR DESCRIPTION
**Pull request by sactage**

### CODE REVIEW

**Comment by TK-999** [comment]
CR ticket: https://wikia-inc.atlassian.net/browse/SUS-373

File: server.js [hunk 1]
```diff
@@ -274,8 +274,8 @@ function authConnection(socket, next){
 
 	var callback = function(data) {
 
-		var usernameOk = (data.username_encoded == name) || (unescape(data.username) == unescape(name))
-		logger.debug("auth test: " + data.username_encoded + "==" + data.username + "==" + unescape(data.username) + "==" + name);
+		var usernameOk = (data.username_encoded == name) || (decodeURIComponent(data.username) == decodeURIComponent(name))
```
Comment (dianafa) [inline server.js]: what about moving logic for determining if username is correct to a separate function and add unit tests for it?

File: WMBridge.js [hunk 2]
```diff
@@ -202,7 +202,7 @@ var clearAuthenticateCache = function(roomId, name) {
 WMBridge.prototype.authenticateUser = function(roomId, name, key, handshake, success, error) {
 	// This cache is only secure because it's checking the .key alongside the roomId/username (roomId
 	// and name can be spoofed, but the forger would not also know the 'key' which MediaWiki generates).
-	var normalizedName = unescape(name).replace(/ /g, '_');
+	var normalizedName = decodeURIComponent(name).replace(/ /g, '_');
 	if(authenticateUserCache[roomId] && authenticateUserCache[roomId][normalizedName]
```
Comment (dianafa) [inline WMBridge.js]: I'd make it separate function as well (see comment below)

File: server.js (hunk 1 above)
Comment (sactage) [inline server.js]: I could do that, sure.

File: WMBridge.js (hunk 2 above)
Comment (sactage) [inline WMBridge.js]: With unit tests as well?

Comment (dianafa) [inline WMBridge.js]: would be great, don't you think? :)

**Comment by sactage** [comment]
@dianafa Is there a good example of how I might introduce a unit test here? I have never done unit testing in nodejs - what library should I use? Chat server code did not have any sort of testing until last week, so there is nothing here for me to follow

If it is more convenient for you, you can message me on slack (`@furry`) to explain - I cannot message you directly without you messaging me first, though, as my slack account is restricted

**Comment by TK-999** [comment]
@sactage it seems to be using the Mocha test framework - https://mochajs.org/

**Comment by dianafa** [comment]
Yes, we've discussed this offline :) Thank you @TK-999 👯

**Comment by TK-999** [comment]
reapplied changes in #34 fixing merge conflict

**Pull request closed without merging**